pkg/util/fs_util.go

```diff
@@ -616,7 +616,7 @@ func CopySymlink(src, dest, buildcontext string) (bool, error) {
 	if err := createParentDirectory(dest); err != nil {
 		return false, err
 	}
-	return false, os.Symlink(link, dest)
+	return CopyFile(link, dest, buildcontext, uid, gid)
```

Comment:
Why the change from link to copy? I'm struggling to connect this change to a flaky test.

Reply:
This test was flaky because of 2 reasons
1. The `link` value was `/tmp/target`. 
   We were linking `kaniko/0/tmp/link` to `/tmp/target` -  should have been `/kaniko/0/tmp/link`
2. The target file `/tmp/target` was getting wiped out before call to `os.Symlink(link, dest)` . - this caused flakiness. 

In the description, i mentioned i fixed the `CopySymlink` to replicate docker behavior. Docker copies the target file to destination link and hence i replaced `os.Symlink` to `CopyFile`.